File: app/src/main/java/loopeer/com/appbarlayout_spring_extension/SpringAppBarLayoutWithTabActivity.java
package loopeer.com.appbarlayout_spring_extension;

import android.os.Bundle;
import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.appbar.AppBarLayoutSpringBehavior;
import com.google.android.material.appbar.CollapsingToolbarLayout;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.appbar.TabScrimHelper;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.github.mmin18.widget.RealtimeBlurView;

public class SpringAppBarLayoutWithTabActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_spring_app_bar_tab_layout);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        CollapsingToolbarLayout collapsingToolbarLayout = findViewById(R.id.toolbar_layout);

        final RealtimeBlurView realtimeBlurView = findViewById(R.id.real_time_blur_view);

        AppBarLayout appBarLayout = findViewById(R.id.app_bar);
        AppBarLayoutSpringBehavior springBehavior = (AppBarLayoutSpringBehavior) ((CoordinatorLayout.LayoutParams) appBarLayout.getLayoutParams()).getBehavior();
        springBehavior.setSpringOffsetCallback(new AppBarLayoutSpringBehavior.SpringOffsetCallback() {
            @Override
            public void springCallback(int offset) {
                int radius = 20 * (240 - offset > 0 ? 240 - offset : 0) / 240;
                realtimeBlurView.setBlurRadius(radius);
            }
        });

        ViewPager viewPager = findViewById(R.id.tabs_viewpager);
        TabLayout tabLayout = findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(viewPager);
        viewPager.setAdapter(new TabFragmentPagerAdapter(getSupportFragmentManager()));
        TabScrimHelper tabScrimHelper = new TabScrimHelper(tabLayout, collapsingToolbarLayout);
        appBarLayout.addOnOffsetChangedListener(tabScrimHelper);
    }
}

File: app/src/main/java/loopeer/com/appbarlayout_spring_extension/SpringAppBarLayoutWithTabActivity.kt
package loopeer.com.appbarlayout_spring_extension

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.viewpager.widget.ViewPager
import com.github.mmin18.widget.RealtimeBlurView
import com.google.android.TabScrimHelper
import com.google.android.material.appbar.AppBarLayout
import com.google.android.material.appbar.AppBarLayoutSpringBehavior
import com.google.android.material.appbar.CollapsingToolbarLayout
import com.google.android.material.tabs.TabLayout

class SpringAppBarLayoutWithTabActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_spring_app_bar_tab_layout)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        val collapsingToolbarLayout = findViewById<CollapsingToolbarLayout>(R.id.toolbar_layout)

        val realtimeBlurView = findViewById<RealtimeBlurView>(R.id.real_time_blur_view)

        val appBarLayout = findViewById<AppBarLayout>(R.id.app_bar)
        val springBehavior = (appBarLayout.layoutParams as CoordinatorLayout.LayoutParams).behavior as AppBarLayoutSpringBehavior?
        springBehavior!!.springOffsetCallback = object : AppBarLayoutSpringBehavior.SpringOffsetCallback {
            override fun springCallback(offset: Int) {
                val radius = 20 * (if (240 - offset > 0) 240 - offset else 0) / 240
                realtimeBlurView.setBlurRadius(radius.toFloat())
            }
        }

        val viewPager = findViewById<ViewPager>(R.id.tabs_viewpager)
        val tabLayout = findViewById<TabLayout>(R.id.tabs)
        tabLayout.setupWithViewPager(viewPager)
        viewPager.adapter = TabFragmentPagerAdapter(supportFragmentManager)
        val tabScrimHelper = TabScrimHelper(tabLayout, collapsingToolbarLayout)
        appBarLayout.addOnOffsetChangedListener(tabScrimHelper)
    }
}
